modules/pallets/messaging-fees/src/types.rs

```diff
@@ -0,0 +1,56 @@
+use codec::{Decode, Encode};
+use ismp::{
+	router::{Request, Response},
+};
+use polkadot_sdk::{
+	frame_support::pallet_prelude::TypeInfo,
+	sp_core,
+	sp_core::RuntimeDebug,
+	sp_runtime::{traits::Zero, DispatchError, Weight},
+};
+use polkadot_sdk::sp_core::sr25519;
+use ismp::host::StateMachine;
+
+pub type AuthorityId = sp_core::sr25519::Public;
+pub type AuthoritySignature = sp_core::sr25519::Signature;
+
+pub enum IncentivizedMessage {
+	Request(Request),
+	Response(Response),
+}
+
+
+#[derive(Clone, Encode, Decode, TypeInfo, PartialEq, Eq, RuntimeDebug)]
+pub struct EpochInfo<BlockNumber> {
+	/// The index of the current epoch
+	pub index: u64,
+	/// The block number at which the epoch started
+	pub start_block: BlockNumber,
+}
+
+impl<BlockNumber: Zero> Default for EpochInfo<BlockNumber> {
+	fn default() -> Self {
+		Self { index: 0, start_block: BlockNumber::zero() }
+	}
+}
+
+/// A trait for Bridge price oracle.
+pub trait PriceOracle<Balance> {
+	fn get_bridge_price() -> Result<Balance, DispatchError>;
```

Comment:
Probably best we use U256 here, the price will be a floating point value.
```suggestion
	fn get_bridge_price() -> Result<U256, DispatchError>;
```